dev-docs: add tracing doc (#1146)
edgelesssys/contrast: dev-docs/coco/tracing.md

# Tracing Kata

> Kata has first-class support for tracing via OpenTelemetry.

Reviewer: We may want to link against the upstream document here: https://github.com/kata-containers/kata-containers/blob/main/docs/tracing.md

# Tracing Kata

Kata has first-class support for tracing via OpenTelemetry. [^1]

> This document describes on how to collect the traces and get a web UI with them.

Reviewer 2: ```suggestion
This document describes how to collect the traces and display them in a web UI.
```

## Enabling tracing in the runtime

The Kata runtime can be configured for tracing by adding the following snippet to
the Kata configuration file (`configuration-qemu-snp.toml`, for example):

```toml
[runtime]
enable_tracing = true
```

The Kata runtime is then configured to send traces to an OpenTelemetry collector,
which needs to be set up separately.

## Running an OpenTelemetry Collector

Running an OpenTelemetry Collector is very simple when using the [Jaeger all-in-one
Docker container](https://www.jaegertracing.io/docs/1.6/getting-started/#all-in-one-docker-image).

> This needs to be started on the runtime's host server (`discovery`, for example):

Reviewer 2: ```suggestion
This needs to be started on the runtime's host server:
```

I believe host server is clear enough and the concrete server name here may be confusing.
